### DISCUSSION TITLE
How to open the wireguard server for other clients in the same LAN?

### DISCUSSION THREAD

**Discussion by viertelb** [pivpn/pivpn#1599] [Q&A] [answered]
It seems that the `pivpn` server with wireguard is blocking connections from clients in the main LAN of the server to the tunnel. Of course I can connect from clients in other LANs over the internet to the wireguard server via the tunnel interfaces. But I also want every client in the LAN of the server to use this server for outgoing traffic - if it is directed to the remote LAN (of the wireguard client).

I therefore added a static route into my router of the server's LAN to redirect the traffic to the wireguard server. 

Now, when I ping on the server I get 

```
pi@RPi64:~ $ ping 192.168.178.1
PING 192.168.178.1 (192.168.178.1) 56(84) bytes of data.
From 10.122.242.1 icmp_seq=1 Destination Host Unreachable
```

Do I need to add something to `iptables`?

To make sure the firewall is not blocking I disabled it before testing. 

The packages reach the wireguard server, but do not go into the `wg0` interface (`10.122.242.1`):

```
 $ traceroute 192.168.178.1
traceroute to 192.168.178.1 (192.168.178.1), 30 hops max, 60 byte packets
 1  192.168.100.1 (192.168.100.1)  0.414 ms  0.338 ms  0.385 ms
 2  192.168.100.90 (192.168.100.90)  1.206 ms  1.215 ms  1.207 ms
 3  * * *
```

**Comment by orazioedoardo** [comment]
Can you draw a sketch of the setup?

**Reply by viertelb** [reply]
@orazioedoardo Please forget about it. I am successfully connecting! I am just not connecting from the PiVPN Server! The server itself does not relay a ping. But all clients can connect! I just did not check this earlier because the ping did not go through.

**Comment by viertelb** [comment, marked as answer]
[Here](https://www.reddit.com/r/WireGuard/comments/krpsc4/wireguard_ping_sendmsg_required_key_not_available/) is a post that has a similar problem. The poster wants to connect from his (server) side tunnel to his friends (clients) side tunnel. He worked it out by adding the client's LAN to allowed IPs:

`AllowedIPs = 10.99.99.5/32, 192.168.190.0/2`

The suggestion was to add the second IP also to the client's side, but I think this was wrong. EDIT: This is definitely wrong, but wrongly suggested so on the wireguard website (Crypto Key section paragraph on the client configuration on this exact page: https://www.wireguard.com/). Source IP is written there, but destination IP should it be.

I did add it to the server, restarted the `wg0` and the client's interface, but nothing changed. 

```
pi@RPi64:~ $ sudo wg
interface: wg0
  public key: <KEY>
  private key: (hidden)
  listening port: 51820

peer: <KEY>
  preshared key: (hidden)
  endpoint: <PUBLIC-IP>:35079
  allowed ips: 10.122.242.2/32, 192.168.178.0/24
  latest handshake: 30 seconds ago
  transfer: 223.69 KiB received, 1.40 MiB sent

```

To break the problem down a bit I am now on the server and try to ping the client.

```
pi@RPi64:~ $ route
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         192.168.100.1   0.0.0.0         UG    202    0        0 eth0
default         192.168.100.1   0.0.0.0         UG    303    0        0 wlan0
10.122.242.0    0.0.0.0         255.255.255.0   U     0      0        0 wg0
192.168.100.0   0.0.0.0         255.255.255.0   U     202    0        0 eth0
192.168.100.0   0.0.0.0         255.255.255.0   U     303    0        0 wlan0
192.168.178.0   0.0.0.0         255.255.255.0   U     0      0        0 wg0
```

I am unable to ping the client's tunnel IP or its native IP:

```
pi@RPi64:~ $ ping 192.168.178.10
PING 192.168.178.10 (192.168.178.10) 56(84) bytes of data.
^C
--- 192.168.178.10 ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 2040ms

pi@RPi64:~ $ ping 10.122.242.2
PING 10.122.242.2 (10.122.242.2) 56(84) bytes of data.
^C

```

**Reply by viertelb** [reply]
I am unable to ping clients in the remote LAN *from the server itself*.

However, I can connect *from the clients*. So, everything works fine!
